content/en/docs/components/pipelines/sdk/v2/build-pipeline.md

```diff
@@ -0,0 +1,522 @@
+}
+a.github-link {
+padding-left: 2.75em;
+background-image: url(/docs/images/logos/github.png);
+background-repeat: no-repeat;
+background-size: auto 75%;
+background-position: left center;
+}
+</style>
+<div class="notebook-links">
+<a class="colab-link" href="https://colab.research.google.com/github/kubeflow/website/blob/master/content/en/docs/components/pipelines/sdk/v2/build-pipeline.ipynb">Run in Google Colab</a>
+<a class="github-link" href="https://github.com/kubeflow/website/blob/master/content/en/docs/components/pipelines/sdk/v2/build-pipeline.ipynb">View source on GitHub</a>
+</div>
+
+
+
+A Kubeflow pipeline is a portable and scalable definition of a machine learning
+(ML) workflow. Each step in your ML workflow, such as preparing data or
+training a model, is an instance of a pipeline component. This document
+provides an overview of pipeline concepts and best practices, and instructions
+describing how to build an ML pipeline.
+
+**Note:** This guide demonstrates how to build pipelines using the Pipelines SDK v2.
+Currently, Kubeflow Pipelines v2 is in development. You can use this guide to start
+building and running pipelines that are compatible with the Pipelines SDK v2.
+
+[Learn more about Pipelines SDK v2][kfpv2].
+
+[kfpv2]: https://www.kubeflow.org/docs/components/pipelines/sdk/v2/v2-compatibility
+
+## Before you begin
+
+1.  Run the following command to install the Kubeflow Pipelines SDK v1.6 or higher.
+    If you run this command in a Jupyter notebook, restart the kernel after
+    installing the SDK. 
+
+
+```python
+$ pip install kfp --upgrade
+```
+
+2.  Import the `kfp` and `kfp.components` packages.
+
+
+```python
+import kfp
+import kfp.components as comp
+import kfp.dsl as dsl
+from kfp.v2 import compiler
+from kfp.v2.dsl import component
+from kfp.v2.dsl import (
+    Input,
+    Output,
+    Artifact,
+    Dataset,
+)
+```
+
+3. Create an instance of the [`kfp.Client` class][kfp-client]. To find your
+   Kubeflow Pipelines cluster's hostname and URL scheme, open the Kubeflow
+   Pipelines user interface in your browser. The URL of the Kubeflow
+   Pipelines user interface is something like 
+   `https://my-cluster.my-organization.com/pipelines`. In this case, the
+   host name and URL scheme are `https://my-cluster.my-organization.com`. 
+
+[kfp-client]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html#kfp.Client
+
+
+```python
+# If you run this command on a Jupyter notebook running on Kubeflow, you can
+# exclude the host parameter.
+# client = kfp.Client()
+client = kfp.Client(host='<your-kubeflow-pipelines-host-name>')
+```
+
+## Understanding pipelines
+
+A Kubeflow pipeline is a portable and scalable definition of an ML workflow,
+based on containers. A pipeline is composed of a set of input parameters and a
+list of the steps in this workflow. Each step in a pipeline is an instance of a
+component, which is represented as an instance of 
+[`ContainerOp`][container-op].
```

Comment:
IMO, its good for users to understand that a step == ContainerOp instance, since they might need to understand how to use the ContainerOp to specify things like GPU requirements in a step.

Reply:
That's a good point!

Reply:
> Somehow I got the impression that ContainerOp is not the only step now. Can you confirm?

Not sure I understand this. A step/task/component(backed by a container implementation) is a `ContainerOp` instance.